PIM_BE/PIMTool/Services/ProjectService.cs

```diff
@@ -0,0 +1,411 @@
+﻿using AutoMapper.Configuration.Conventions;
+using Microsoft.EntityFrameworkCore;
+using PIMTool.Core.Constants;
+using PIMTool.Core.Domain.Entities;
+using PIMTool.Core.Interfaces.Repositories;
+using PIMTool.Core.Interfaces.Services;
+using PIMTool.Dtos;
+using PIMTool.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Dynamic.Core;
+using System.Text;
+using PIMTool.Core.Domain.Objects;
+
+namespace PIMTool.Services;
+
+public class ProjectService : IProjectService
+{
+    private readonly IRepository<Project> _repository;
+    private readonly IRepository<Employee> _repoEmp;
+    private readonly string DES = "DES";
+    private readonly string ASC = "ASC";
+    public ProjectService(IRepository<Project> repository, IRepository<Employee> repoEmp)
+    {
+        _repository = repository;
+        _repoEmp = repoEmp;
+    }
+
+    public async Task Create(Project project)
+    {
+        IEnumerable<Project> projectEnumerator = await _repository.GetAll();
+        bool check = project.UniqueProjectNumber(projectEnumerator);
```

Comment:
[C] it is very bad which it load all data with  IEnumerable<Project> projectEnumerator = await _repository.GetAll();